DigitalLearningSolutions.Data/DataServices/SupervisorService.cs

```diff
@@ -156,7 +156,10 @@ public int AddSuperviseDelegate(int? supervisorAdminId, int? delegateUserId, str
                             FROM Users u
                             INNER JOIN DelegateAccounts da
                             ON da.UserID = u.ID
-                            WHERE PrimaryEmail = @delegateEmail
+	                        INNER JOIN UserCentreDetails ucd
+	                        ON ucd.UserID = u.ID
+                            AND ucd.CentreID = da.CentreID
+                            WHERE ucd.Email = @delegateEmail

```

Comment:
Should this be `WHERE (ucd.Email = @delegateEmail OR u.PrimaryEmail = @delegateEmail)` ?

Reply:
Yes that seems a more robust approach thanks.